mllib/src/test/scala/org/apache/spark/ml/util/MLTestingUtils.scala

```diff
@@ -117,7 +117,7 @@ object MLTestingUtils extends SparkFunSuite {
       Seq(ShortType, LongType, IntegerType, FloatType, ByteType, DoubleType, DecimalType(10, 0))
     types.map { t =>
         val castDF = df.select(col(labelColName).cast(t), col(featuresColName))
-        t -> TreeTests.setMetadata(castDF, 2, labelColName, featuresColName)
+        t -> TreeTests.setMetadata(castDF, 0, labelColName, featuresColName)
```

Comment:
What is this for? If the intent is to force `getNumClasses` to infer the number of classes, then you're no longer testing the not inferred case. Further, the point of this PR is to eliminate the need to do that since it is not a robust solution, IMO. 

Also, I'd like to remove the dependence on `TreeTests` here (and `genRegressionDF`) and just explicitly set the attributes in the functions.